includes/log-handlers/class-wc-log-handler-file.php

```diff
@@ -0,0 +1,218 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+if ( ! class_exists( 'WC_Log_Handler' ) ) {
+	include_once( dirname( dirname( __FILE__ ) ) . '/abstracts/abstract-wc-log-handler.php' );
+}
+
+/**
+ * Handles log entries by writing to a file.
+ *
+ * @class          WC_Log_Handler_File
+ * @version        1.0.0
+ * @package        WooCommerce/Classes
+ * @category       Class
+ * @author         WooThemes
+ */
+class WC_Log_Handler_File extends WC_Log_Handler {
+
+	/**
+	 * Stores open file handles.
+	 *
+	 * @var array
+	 * @access private
+	 */
+	private $handles = array();
+
+	/**
+	 * Destructor.
+	 */
+	public function __destruct() {
+		foreach ( $this->handles as $handle ) {
+			if ( is_resource( $handle ) ) {
+				fclose( $handle );
+			}
+		}
+	}
+
+	/**
+	 * Handle a log entry.
+	 *
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 * @param string $message
+	 * @param array $context {
+	 *     Array with additional information
+	 *
+	 *     @type string $tag Optional. The tag will be used to determine which file an entry will be written to.
+	 * }
+	 *
+	 * @return bool log entry should bubble to further loggers.
+	 */
+	public function handle( $level, $timestamp, $message, $context ) {
+
+		if ( ! $this->should_handle( $level ) ) {
+			return true;
+		}
+
+		if ( isset( $context['tag'] ) && $context['tag'] ) {
+			$handle = $context['tag'];
+		} else {
+			$handle = 'log';
+		}
+
+		$entry = $this->format_entry( $level, $timestamp, $message, $context );
+
+		// Bubble if add is NOT successful
+		return ! $this->add( $entry, $handle );
+	}
+
+	/**
+	 * Builds a log entry text from level, timestamp and message.
+	 *
+	 * Attempt to ensure backwards compatibility for legacy WC_Logger::add calls
+	 *
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 * @param int $timestamp log entry timestamp
+	 * @param string $message provided log message
+	 * @param array $context {
+	 *     Array with additional information
+	 *
+	 *     @type string $tag Optional. The tag will be used to determine which file an entry will be written to.
+	 * }
+	 *
+	 * @return string Formatted log entry
+	 */
+	public function format_entry( $level, $timestamp, $message, $context ) {
+
+		if ( isset( $context['_legacy'] ) && true === $context['_legacy'] ) {
+			if ( isset( $context['tag'] ) && $context['tag'] ) {
+				$handle = $context['tag'];
+			} else {
+				$handle = 'log';
+			}
+			$message = apply_filters( 'woocommerce_logger_add_message', $message, $handle );
+			$time = date_i18n( 'm-d-Y @ H:i:s -' );
```

Comment:
I maintained the format here for compatibility with legacy `->add` calls. Calls to the updated `->log`, `->debug`, etc. are formatted with `c` (ISO 8601 date 2004-02-12T15:19:21+00:00). I found that format in a logger spec and it seemed appropriate. I'm open to discuss different alternatives.